frontends/p4/typeMap.cpp

```diff
@@ -46,6 +46,28 @@ bool TypeMap::typeIsEmpty(const IR::Type *type) const {
     return false;
 }
 
+bool TypeMap::typeIsFixedWidth(const IR::Type *type) const {
```

Comment:
The interpreter defaulted to true for most base types, but I am not sure whether that is a good idea for this case here. Looking at 
```
baseType
    : BOOL
    | MATCH_KIND
    | ERROR
    | BIT
    | STRING
    | INT
    | BIT "<" INTEGER ">"
    | INT "<" INTEGER ">"
    | VARBIT "<" INTEGER ">"
    | BIT "<" "(" expression ")" ">"
    | INT "<" "(" expression ")" ">"
    | VARBIT "<" "(" expression ")" ">"
    ;
```
I'd consider only `Type_Boolean` and `Type_Bits` fixed-width types. We should return true for just these two types.

Now whether this distinction makes sense for lookahead in particular can be argued, but since this is a core type map function it should be safer.